### PR TITLE
Deduplicate with latest workspace feature

### PR DESCRIPTION
**Pull request by boundless-forest**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -4,3 +4,49 @@ members = [
...
+sp-core                    = { default-features = false, git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.37" }
+sp-finality-grandpa        = { default-features = false, git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.37" }
+sp-io                      = { default-features = false, git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.37" }
+sp-runtime                 = { default-features = false, git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.37" }
+sp-state-machine           = { default-features = false, git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.37" }
+sp-std                     = { default-features = false, git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.37" }
+sp-trie                    = { default-features = false, git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.37" }
+sp-version                 = { default-features = false, git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.37" }
```
Comment (jiguantong) [inline Cargo.toml:52]: Empty line

File: Cargo.toml [hunk 2]
```diff
@@ -4,3 +4,49 @@ members = [
 	"primitives/*",
 	"runtime-common",
 ]
+
+[workspace.dependencies]
```
Comment (jiguantong) [inline Cargo.toml:8]: Keep the same indentation format as darwinia2.0 repo?

Comment (boundless-forest) [reply Cargo.toml:8]: Let's keep this format and update the darwinia 2.0 repo.

File: Cargo.lock [hunk 3]
```diff
@@ -107,6 +107,12 @@ version = "4.1.0"
 source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "6a913633b0c922e6b745072795f50d90ebea78ba31a57e2ac8c2fc7b50950949"
 
+[[package]]
```
Comment (jiguantong) [inline Cargo.lock:110]: Are these changes expected?

Comment (boundless-forest) [reply Cargo.lock:110]: I upgrade the `array-bytes` version to the darwinia 2.0 version.

Some dep versions are different in different crates, now they are the same.

Comment (jiguantong) [reply Cargo.lock:110]: So the [`array-bytes-4.1.0`](https://github.com/darwinia-network/darwinia-messages-substrate/blob/b5be9e1a78e7408ffc67285603886d6548a48319/Cargo.lock#L105) should be removed.

Comment (boundless-forest) [reply Cargo.lock:110]: Great catch:

```toml
[[package]]
name = "sp-core"
version = "7.0.0"
source = "git+https://github.com/paritytech/substrate?branch=polkadot-v0.9.37#f38bd6671d460293c93062cc1e4fe9e9e490cb29"
dependencies = [
 "array-bytes 4.2.0",
 "base58",
 "bitflags",
 "blake2",
 "dyn-clonable",
 "ed25519-zebra",
 "futures",
 "hash-db",
 "hash256-std-hasher",
 "impl-serde 0.4.0",
 "lazy_static",
 "libsecp256k1",
 "log",
 "merlin",
 "parity-scale-codec",
 "parking_lot",
 "primitive-types",
 "rand 0.8.5",
 "regex",
 "scale-info",
 "schnorrkel",
 "secp256k1",
 "secrecy",
 "serde",
 "sp-core-hashing",
 "sp-debug-derive",
 "sp-externalities",
 "sp-runtime-interface",
 "sp-std",
 "sp-storage",
 "ss58-registry",
 "substrate-bip39",
 "thiserror",
 "tiny-bip39",
 "zeroize",
]
```
The old version comes from the substrate crates. So I think it should be fixed there. @AurevoirXavier

**Review by jiguantong — APPROVED** [review]

**Pull request merged**